apps/rush-lib/src/logic/pnpm/PnpmProjectDependencyManifest.ts

```diff
@@ -211,4 +236,32 @@ export class PnpmProjectDependencyManifest {
       }
     }
   }
+
+  /**
+   * The version specifier for a dependency can sometimes come in the form of
+   * '{semVer}_peerDep1@1.2.3+peerDep2@4.5.6'. This is parsed and returned as a dictionary mapping
+   * the peer dependency to it's appropriate PNPM dependency key.
+   */
+  private _parsePeerDependencyKeysFromSpecifier(specifier: string): { [peerDependencyName: string]: string } {
+    const parsedPeerDependencyKeys: { [peerDependencyName: string]: string } = {};
+
+    const split1: RegExpExecArray | null = /^[^_]+_(.+)$/.exec(specifier);
+    if (split1) {
+      const combinedPeerDependencies: string = split1[1];
+      // Parse "eslint@6.6.0+typescript@3.6.4" --> ["eslint@6.6.0", "typescript@3.6.4"]
```

Comment:
What happens if the package name has a scope like `@rushstack/eslint-config` instead of `eslint`?

Reply:
We looked into this and found that the version specifier then becomes a commit hash. It actually existed in the repro posted in the original issue however it didn't seem to cause an issue. The regex is designed such that it would ignore this style so, if this problem does still exist, the new code wont harm it at the very least. For now, we're going to say this isn't an issue and will revisit if it becomes one.

Reply:
Agreed. 👍 

@zkochan I wonder if the `pnpm-lock.yaml` file format could be improved in this edge case.  It appears that a fairly complex algorithm is required to determine what versions PNPM chose for the peer dependencies. It would be great if this information was represented more explicitly in the file.

Reply:
all that information is present. Such packages have a peer dependencies field with the specs of wanted peer deps. Also, the resolved peers are in the dependencies field.

Reply:
https://github.com/pnpm/pnpm/blob/68e444ea2959aecc82e387c1e83b3f8649a14d01/pnpm-lock.yaml#L2168-L2176

Reply:
@zkochan consider this example:
```yaml
  /@typescript-eslint/typescript-estree/2.6.1_typescript@3.6.4:
    dependencies:
      debug: 4.1.1
      glob: 7.1.6
      is-glob: 4.0.1
      lodash.unescape: 4.0.1
      semver: 6.3.0
      tsutils: 3.17.1_typescript@3.6.4
    dev: false
    engines:
      node: ^8.10.0 || ^10.13.0 || >=11.10.1
    peerDependencies:
      typescript: '*'
    peerDependenciesMeta:
      typescript:
        optional: true
```
As far as we understand, the only easy way to know that `typescript: '*'` was resolved as `3.6.4` is by parsing `2.6.1_typescript@3.6.4`.

Reply:
Here's another one with non-optional peers:
```yaml
 /@typescript-eslint/experimental-utils/2.7.0_eslint@6.6.0+typescript@3.6.4:
    dependencies:
      '@types/json-schema': 7.0.3
      '@typescript-eslint/typescript-estree': 2.7.0_typescript@3.6.4
      eslint: 6.6.0
      eslint-scope: 5.0.0
    dev: false
    engines:
      node: ^8.10.0 || ^10.13.0 || >=11.10.1
    peerDependencies:
      eslint: '*'
      typescript: '*'
    resolution:
      integrity: sha512-9/L/OJh2a5G2ltgBWJpHRfGnt61AgDeH6rsdg59BH0naQseSwR7abwHq3D5/op0KYD/zFT4LS5gGvWcMmegTEg==
```

We know that `eslint: '*'` was resolved as 6.6.0 only by parsing `2.7.0_eslint@6.6.0+typescript@3.6.4`.

Reply:
Seems like these packages don't have typescript as their actual peer dependency. typescript is a peer dependency of their dependency. So if you check `/tsutils/3.17.1_typescript@3.6.4`, it should have typescript in dependencies